### ISSUE TITLE
SAM file location

### ISSUE THREAD

**Issue by devilock76** [FreeRDP/FreeRDP#3434]
I was wondering if there is a way to specify another location for the SAM file than /etc/winpr.  I would like to put mine in my home directory instead of /etc.  So far the only option I can come up with is creating a sym link from the original location to the one in my home.  But I don't like that as a solution at the moment.

**Comment by akallabeth** [comment]
Currently this is hard coded, so no way to change without recompiling.
Of course we always appreciate pull requests ;)

**Comment by devilock76** [comment]
My C skills are a bit lacking these days, but noted.  Let me ask another question.  When I first started using xfreerdp the only way I could negotiate a TS gateway was by creating the SAM file.  But is that really a requirement?  In short must there be a SAM file or have I taken it for granted that that was the solution to whatever else I might have been doing wrong?

**Comment by akallabeth** [comment]
Looks like the SAM file is not really used at all (that is assuming you're using latest nightly builds)

**Comment by bmiklautz** [comment]
@devilock76 if you using xfreerdp as client there a SAM file shouldn't be required. For gateway there are command line parameters.

**Comment by devilock76** [comment]
I am using stable 1.1 at the moment.  I think perhaps the reason I needed it is I was trying to use just one password field /p and that it would hit both, and then I think somewhere in there I tried both...  At the time I was new to using it and trying to get set up fast at one point I got the SAM error, google-fu'd it and got to the point of it was working and said done.  I will do some experimentation again on stable since that is what I have and then go from there.

Oh my usage is client only.  Basically I am the only oddball in my office trying to use linux to connect to windows desktops when remote.  We have an embedded windows culture there that has been hard to break.  If I had my way I would not have to do this at all.  But I am thankful for this project as a solution because this is about the only way from linux to negotiate our TS Gateway setup.

Thank you for your time in answering my questions.
